src/services/document-service.js

```diff
@@ -78,16 +80,17 @@ const createNewDocumentRevision = ({ doc, documentKey, userId, nextOrder, restor
     _id: uniqueId.create(),
     key: documentKey,
     order: nextOrder,
-    restoredFrom,
+    restoredFrom: restoredFrom || '',
     createdOn: new Date(),
-    createdBy: userId,
+    createdBy: userId || '',
     title: doc.title || '',
     slug: doc.slug || '',
-    namespace: doc.namespace,
-    language: doc.language,
+    namespace: doc.namespace || '',
+    language: doc.language || '',
     sections: sections || doc.sections,
-    tags: doc.tags,
-    archived: doc.archived || false
+    tags: doc.tags || [],
+    archived: doc.archived || false,
+    origin: doc.origin || DOCUMENT_ORIGIN.internal
```

Comment:
There's a test that ensures the projection has the required fields, maybe you can update that :). It doesn't look for all the stuff that shouldn't be there, just that the sections don't get mapped that's why it didn't fail.